content/docs/pipelines/sdk/manipulate-resources.md

```diff
@@ -0,0 +1,275 @@
+
+---
+
+## Volume Snapshots
+
+Request the creation of Volume Snapshot instances simple and fast.
+
+### VolumeSnapshotOp
+
+A ResourceOp specialized in Volume Snapshot creation.
+
+[Link](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.VolumeSnapshotOp)
+to the corresponding Python library.
+
+**NOTE:** You should check if your Kubernetes cluster admin has Volume Snapshots
+enabled in your cluster.
+
+#### Arguments
+
+The following arguments are an extension to the `ResourceOp` arguments.
+If a `k8s_resource` is passed, then none of the following may be provided.
+
+* `resource_name`: The name of the resource which will be created.
+  This string will be prepended with the workflow name.
+  This may contain `PipelineParam`s.
+  (_required_)
+* `pvc`: The name of the PVC to be snapshotted.
+  This may contain `PipelineParam`s.
+  (_optional_)
+* `snapshot_class`: The snapshot storage class to be used.
+  This may contain `PipelineParam`s.
+  (_optional_)
+* `volume`: An instance of a `V1Volume`, or its inherited type (e.g.
+  `PipelineVolume`).
+  This may contain `PipelineParam`s.
+  (_optional_)
+* `annotations`: Annotations to be patched in the `VolumeSnapshot`.
+  These may contain `PipelineParam`s.
+  (_optional_)
+
+**NOTE:** One of the `pvc` or `volume` needs to be provided.
+
+#### Outputs
+
+Additionally to the whole specification of the resource and its name
+(`ResourceOp` defaults), a `VolumeSnapshotOp` also outputs the `restoreSize` of
+the bounded `VolumeSnapshot` (as `step.outputs["size"]`).
+This is the minimum size for a PVC clone of that snapshot.
+
+#### Useful information
+
+`VolumeSnapshotOp` steps have a `.snapshot` attribute which is a
+`V1TypedLocalObjectReference`.
+This can be passed as a `data_source` to create a PVC out of that
+`VolumeSnapshot`.
+The user may otherwise use the `step.outputs["name"]` as `data_source`.
+
+### Samples
+
+For better understanding, please refer to the following samples:
+[1](https://github.com/kubeflow/pipelines/blob/master/samples/core/volume_snapshot_ops/volume_snapshot_op.py),
+[2](https://github.com/kubeflow/pipelines/blob/master/samples/contrib/volume_snapshot_ops/volume_snapshotop_rokurl.py)
+
+## Next steps
+
+* See samples in Kubeflow Pipelines 
+  [repository](https://github.com/kubeflow/pipelines/tree/master/samples).
+  For instance, check these samples of 
+  [ResourceOps](https://github.com/kubeflow/pipelines/tree/master/samples/core/resource_ops), 
+  [VolumeOps](https://github.com/kubeflow/pipelines/tree/master/samples/core/volume_ops)
+  and 
+  [VolumeSnapshotOps](https://github.com/kubeflow/pipelines/tree/master/samples/core/volume_snapshot_ops).
+* Learn more about the 
+  [Kubeflow Pipelines domain-specific language (DSL)](/docs/pipelines/sdk/dsl-overview/),
+  a set of Python libraries that you can use to specify ML pipelines.
+* For quick iteration, 
+  [build components and pipelines](/docs/pipelines/sdk/build-components/).
```

Comment:
Heh, just one issue in this PR and it's on the very last line. :) The link is broken - remove the "s" at the end of "components".

Reply:
Oops! Sorry for that, fixed it.